### PR TITLE
Joel/separate stories from page

### PR DESCRIPTION
**Pull request by jpeters5392**
TL;DR

Each component now gets a page that imports a different MDX file with the story info.  The reason for this is Refinitiv wants to render the story list for designers, but then developers will see the story list mixed in with instructions.  There are now 2 templates for stories.  The plain list and then the mixed-in list.

### CODE REVIEW

**Comment by jpeters5392** [comment]
See https://github.com/hydrateio/emcm-ui/pull/2 for consuming implementation

**Comment by transitive-bullshit** [comment]
This looks fine for Refinitiv, but it seems unrelated to the open source version of `docz-plugin-storybook`. Do you think it makes sense to just use a fork of `docz-plugin-storybook` for Refinitiv and keep the OSS version agnostic to client-specific changes?

**Comment by jpeters5392** [comment]
@transitive-bullshit Take a look at my updates.  Now there shouldn't be any Refinitiv specific pieces in here, although returning an array from `storiesOfTemplate` gives consumers a little extra flexibility

File: src/react/Story.js [hunk 1]
```diff
@@ -26,6 +28,16 @@ export default class Story extends Component {
       )
     }
 
+    if (usageInstructions) {
+      const UsageInstructions = usageInstructions
+      return (
+        <React.Fragment>
+          <StoryWrapper {...rest}>{story()}</StoryWrapper>
+          <UsageInstructions />
```
[the remark this answers is not in the document]
Comment (jpeters5392) [reply src/react/Story.js]: Oh right, this is still here.  I was going to abstract this out as well.

File: src/index.js [hunk 2]
```diff
@@ -13,6 +13,12 @@ const minStorybookSemver = '>=3'
 const v3StorybookSemver = '^3'
 const v4StorybookSemver = '^4'
 
+const ensureArray = (value) => {
+  if (value instanceof Array) return value
```
Comment (transitive-bullshit) [inline src/index.js]: this should be `Array.isArray`. don't ask me why it's better as i have no idea haha 😄 i just know it's preferred in the js world from what i've seen

File: src/react/Story.js [hunk 3]
```diff
@@ -7,13 +7,15 @@ import { _clientAPI } from '../shim'
 export default class Story extends Component {
   static propTypes = {
     kind: PropTypes.string.isRequired,
-    name: PropTypes.string.isRequired
+    name: PropTypes.string.isRequired,
+    usageInstructions: PropTypes.node
```
Comment (transitive-bullshit) [inline src/react/Story.js]: i really don't think `usageInstructions` belongs on a `Story`.

this would be better handled by the custom story template, no?

**Comment by jpeters5392** [comment]
Apparently it is preferred because it works down into iframes.  I ran into
a similar issue with react-frame-component I think, where if you attempt to
check if something is an instance of HTMLElement inside of the iframe since
those browser interfaces are unique per document, so the nodes in the
iframe are instances of the iframe's HTMLElement and not the parent
document where the code is run.

**Comment by transitive-bullshit** [comment]
any idea why CI is failing?

**Comment by jpeters5392** [comment]
It looks like it might be failing to build the tests.  I'll give that a look

**Comment by jpeters5392** [comment]
I had to add `@babel/runtime` as a dev dependency.  It looks like our tests also build the examples so I need to go through those and make sure that they all still work.  I'm flying tomorrow so I'll get back to this later in the week.

**Pull request merged**